tools/marketplace/helper/bundleHelper.go

```diff
@@ -0,0 +1,311 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package helper
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	yaml2 "github.com/ghodss/yaml"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/operator-framework/operator-marketplace/pkg/datastore"
+	v1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+
+	"k8s.io/klog"
+)
+
+//Channel from which OLM bundle is consumed
+type Channel struct {
+	Name       string `yaml:"name"`
+	CurrentCSV string `yaml:"currentCSV"`
+}
+
+//Pkg - Package to be consumed
+type Pkg struct {
+	PkgName  string    `yaml:"packageName"`
+	Channels []Channel `yaml:"channels"`
+}
+
+//Data - OLM bundle wrapper
+type Data struct {
+	CSVs     string `yaml:"clusterServiceVersions"`
+	CRDs     string `yaml:"customResourceDefinitions"`
+	Packages string `yaml:"packages"`
+}
+
+//Bundle - OLM bundle
+type Bundle struct {
+	Data Data `yaml:"data"`
+}
+
+//BundleHelper - object that provides logic of fetching OLM bundle from marketplace quay repo
+type BundleHelper struct {
+	repo      string
+	namespace string
+	Pkgs      []Pkg
+	CRDs      []v1beta1.CustomResourceDefinition
+	CSVs      []yaml.MapSlice
+}
+
+//NewBundleHelper - Ctor
+func NewBundleHelper(repo, namespace string) (*BundleHelper, error) {
+	bh := &BundleHelper{
+		repo:      repo,
+		namespace: namespace,
+	}
+	if err := bh.downloadAndParseBundle(); err != nil {
+		klog.Errorf("Failed to  download bundle", err)
+		return nil, err
+	}
+	return bh, nil
+}
+
+func (bh *BundleHelper) downloadAndParseBundle() error {
+
+	// prepare the app registry client
+	options := appregistry.Options{
+		Source:    "https://quay.io/cnr",
+		AuthToken: "", // not needed for public applications
+	}
+	client, err := appregistry.NewClientFactory().New(options)
+	if err != nil {
+		klog.Errorf("Failed to  create new appregistry client", err)
+		return err
+	}
+
+	// get latest bundle info
+	bundles, err := client.ListPackages(bh.namespace)
+	if err != nil {
+		klog.Errorf("Failed to  list packages", err)
+		return err
+	}
+
+	if len(bundles) == 0 {
+		klog.Infof("no new bundles")
```

Comment:
I would use Info or Infoln if I am not using arguments in the message.

Reply:
Done